pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -363,6 +372,31 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
 	return result, nil
 }
 
+// Record reboot requests
+func checkUpdatedRebootRequest(host *metal3v1alpha1.BareMetalHost) (dirty bool) {
```

Comment:
Should we refuse to reboot a host that hasn't been provisioned? Or do we want to allow this as an option for resolving issues with inspection?

Reply:
Good question.
If the host has _never_ been provisioned then LastBooted won't be set, so we won't try to reboot.
However, if the host has been provisioned and then deprovisioned I think we will try to reboot it. If so that's an inconsistency that I need to fix.

Reply:
It feels like asking for a power cycle can be independent of having an image provisioned to the host. At the very least we need to support externally provisioned hosts, but I don't see any particular reason to restrict it at all. If we prioritize processing a reboot request so it happens before any of the other operations like provisioning or profile matching, then it might work as remediation for failure of some other operation.